### ISSUE TITLE
vehicle's owner issue

### ISSUE THREAD

**Issue by Mikk155** [ZehMatt/Lambda#343]
![20241220184712_1](https://github.com/user-attachments/assets/4ee4c543-e994-4e54-ba5c-9541f4f0f1a1)
To simplify, let's call the nearest vehicle A and the other where my friend is B
A was originaly mine, i died and a friend of me was able to use A and move it away (for propertly revive me as i was stuck in it) then he was unable to enter B but on his end his vehicle A, as the screen shows it was also mine but on my end i was unable to enter on any of them, i haven't really checked the code of it at all but i'd suspect this would be a very rare case of conditions, as far as i recall i just died and he used my vehicle

**Comment by ZehMatt** [comment]
Ordinarily when someone takes over a vehicle because the previous owner died should normally clear the other vehicles, I might have missed something somewhere so I'll have to look into it.

**Comment by Mikk155** [comment]
Yes. It seemed to work lile you described in all of my runs except for that one. i do think that when i died, My vehicle was accessible for other players and there the logic goes off. leaving that player unable to enter the vehicle but the sprite "your vehicle" says contrary.

I didn't tried to replicate but here is some more info if you manage to do

- Player A die
- player B enters player A's vehicle
- player B revive player A
- player B try to enter his vehicle

There were 3 players connected in the server. If that's of some use.

**Comment by ZehMatt** [comment]
Ah I think the issue might be the reviving part which probably skips some of the calls that ordinarily happen when spawning. It is currently not a huge priority to me but I will see to fix it eventually, thanks for the information.
